Account page updates

nenadV91 opened the pull request:
# Summary

Adds some design updates

Desktop
![Screenshot from 2022-08-16 12-44-40](https://user-images.githubusercontent.com/34926005/184861337-c701349a-2923-4fde-94c3-498d3284b7f6.png)

Tablet
![Screenshot from 2022-08-16 12-44-58](https://user-images.githubusercontent.com/34926005/184861367-83c3ffdf-8351-40e9-8969-f2cffff7fd47.png)

Mobile
![Screenshot from 2022-08-16 12-45-10](https://user-images.githubusercontent.com/34926005/184861388-9b787013-8132-42d2-a896-b25ff6bab8ec.png)

For the issues here https://github.com/cowprotocol/cowswap/issues/844
1. tried with 1 inch address, but couldn't reproduce, maybe this is fixed now
2. this seems to also work now
3. fixed
4. fixed
5. fixed
6. fixed
7. In my opinion, this is fine

- Fixes card loader on Balance page
- https://github.com/cowprotocol/cowswap/pull/657#pullrequestreview-1053824234 Should be fixed now

Also there is a small change in this PR for the unsupported network text suggested by @elena-zh

shoom3301 commented on src/custom/components/Popups/FailedNetworkSwitchPopupMod.tsx:
```diff
@@ -23,18 +27,23 @@ export default function FailedNetworkSwitchPopup({
+  const getErrorMessage = useCallback(() => {
+    return isUnsupportedNetwork
+      ? 'Please connect your wallet to one of our supported networks: ' +
+          NETWORK_SELECTOR_CHAINS.map((chainId) => getChainInfo(chainId)?.label)
```
Should we use `@lingui/macro` here?

shoom3301 replied:
Oh, I see we use the result in the `<Trans>`

shoom3301 commented on src/custom/components/Popups/FailedNetworkSwitchPopupMod.tsx:
```diff
@@ -23,18 +27,23 @@ export default function FailedNetworkSwitchPopup({
   const chainInfo = CHAIN_INFO[chainId]
   const theme = useContext(ThemeContext)
 
+  const getErrorMessage = useCallback(() => {
```
`useMemo()` looks more suitable for these purposes

shoom3301 commented on src/custom/components/Tokens/TokensTable.tsx:
```diff
@@ -101,12 +92,29 @@ export default function TokenTable({
       return rawTokensData
     }
-  }, [rawTokensData, debouncedQuery])
+
```
Let's create a separate hook from this useMemo()

shoom3301 commented on src/custom/pages/Account/Tokens/TokensOverview.tsx:
```diff
@@ -110,31 +156,54 @@ export default function TokensOverview() {
+          <TokenSearchInput
+            type="text"
+            id="token-search-input"
+            placeholder={`Search name/symbol or paste address`}
```
Should we use `@lingui/macro` here?

shoom3301 commented on src/custom/pages/Account/Balances.tsx:
```diff
@@ -207,92 +207,88 @@ export default function Profile() {
                 <span>
-                  <i>Available COW balance</i>
-                  <b title={`${cowBalanceMax} COW`}>{cowBalance} COW</b>
+                  <i>Total vCOW balance</i>
```
Should we use `<Trans>` here?

elena-zh commented:
Hey @nenadV91 , great PR!

Some issues I have found: 
1. The app is crashed when I switch to a not supported network on the Tokens page
![crash when unsupported](https://user-images.githubusercontent.com/70885163/184914635-8bb83147-ab61-446d-bc7e-61aadcace7f0.jpg)
2. What is default sorting of tokens on the list? 
3. I can face some issues with Balances column alignment in a tablet view (plus Balance column looks to close to token name)
![alignment](https://user-images.githubusercontent.com/70885163/184914894-5bf13472-d74f-40ae-a380-4636def6af62.jpg)
4. And balance values can be truncated in tablet/mobile views when a balance is too big
![balance truncated](https://user-images.githubusercontent.com/70885163/184915067-7a4c8e35-f16e-4c3b-9219-383cb7c8ed39.jpg)
5. Seems that sorting by balance is not correctly working. See the image( from Gnosis chain)
![sorting](https://user-images.githubusercontent.com/70885163/184915401-58cfd606-1234-498d-9940-f0c741e7ffae.jpg)
6. When I search for 't', I see a bunch of tokens that do not contain 't' in name (maybe in address). Maybe we should prioritize search results somehow to show tokens that start with 't', then contain 't' ('t' I use as an example :) ) , than everything that contains 't' in the address. WDYT?
![search for t](https://user-images.githubusercontent.com/70885163/184919135-87bb04cd-0168-411f-8a43-24aebbe4d025.jpg)

7. Maybe we should center-align list view dropdown in a mobile view due to is exceeds screen width in certain devices?
![image](https://user-images.githubusercontent.com/70885163/184919549-f68badbd-7486-4b34-aadd-37c6852fcccc.png)

Thanks!

anxolin reviewed (commented):
Can be done in another PR, but we should add the menu items:

<img width="945" alt="image" src="https://user-images.githubusercontent.com/2352112/184931989-7205ad64-6fa2-4b84-a58b-db2be50c2702.png">


The account feels a bit sad with the current content. Maybe is fine for now, but nice to add the list of transactions. Your history and trades
<img width="1787" alt="image" src="https://user-images.githubusercontent.com/2352112/184932297-cfae9a19-4f68-43ae-907b-a3b4df359d29.png">

I guess adding the orders, or a table with the history is not low hanging, right?
<img width="722" alt="image" src="https://user-images.githubusercontent.com/2352112/184932797-575ee3e8-62b9-4b83-bb78-f6ee4fcfd93b.png">


It could be something like this alternatively (if easier): https://protocol-explorer.dev.gnosisdev.com/gc/address/0x424a46612794dbb8000194937834250Dc723fFa5 


Also, this is huuuge, was this intended?
<img width="1780" alt="image" src="https://user-images.githubusercontent.com/2352112/184932667-854a8e4e-7125-497c-a987-cce3a8a3c2da.png">

elena-zh commented:
@anxolin 
> Also, this is huuuge, was this intended?

It is reported in #854 task. 

As for the rest you notes, I opened separate tasks for them: 
 - https://github.com/cowprotocol/cowswap/issues/940
 - https://github.com/cowprotocol/cowswap/issues/939

Thanks!

nenadV91 commented:
@elena-zh I've added updates for these points
1. should be fixed now
2. its the same as in the swap modal
3. fixed
4. fixed
5. I'm just going to disable this for now
6. I like the suggestion but I think we can skip this for now
7. fixed

nenadV91 commented:
@anxolin Good points, but outside of scope of this PR.
I would like to merge this and then we can decide what to next for other parts.

alfetopito commented on src/custom/components/Popups/FailedNetworkSwitchPopupMod.tsx:
```diff
@@ -43,7 +42,7 @@ export default function FailedNetworkSwitchPopup({
       <AutoColumn gap="8px">
         <ThemedText.Body fontWeight={500} color={isUnsupportedNetwork ? theme.text2 : theme.text1}>
-          <Trans>{getErrorMessage()}</Trans>
+          <Trans>{getErrorMessage}</Trans>
```
Rename `getErrorMessage` to `errorMessage`?

nenadV91 replied:
Nice catch

alfetopito commented on src/custom/components/Tokens/TokensTable.tsx:
```diff
@@ -26,14 +25,11 @@ import { useErrorModal } from 'hooks/useErrorMessageAndModal'
 import { useToggleWalletModal } from 'state/application/hooks'
 import usePrevious from 'hooks/usePrevious'
-import { useWeb3React } from '@web3-react/core'
+// import { useWeb3React } from '@web3-react/core'
```
Not a mod, might as well remove rather than comment it out

alfetopito commented on src/custom/components/Tokens/TokensTable.tsx:
```diff
@@ -246,7 +219,7 @@ export default function TokenTable({
                 <Trans>Name {arrow(SORT_FIELD.NAME)}</Trans>
               </ClickableText>
-              <ClickableText disabled={!account} onClick={() => (account ? handleSort(SORT_FIELD.BALANCE) : false)}>
+              <ClickableText disabled={true} /* onClick={() => (account ? handleSort(SORT_FIELD.BALANCE) : false)} */>
```
Any reason to keep the comment?
Otherwise:

```suggestion
              <ClickableText disabled>
```

nenadV91 replied:
I've commented it out to disable the sorting on Balance for now, but we will probably use this in the future.

elena-zh commented:
Hey @nenadV91 , great changes!

I have created separate issue for case 5 #947 and 6 #946  to address these issues later. 

As for the current implementation: 
1. Still it would be nice to add a colon to the message in the Account modal when connected to an unsupported network
![add a colon](https://user-images.githubusercontent.com/70885163/185918013-8b496613-d969-40ad-b7b5-4dfefed40789.jpg)
2. I would add the same message as it it on the Account modal to the Tokens table when connected to an unsupported network: 
![the same message](https://user-images.githubusercontent.com/70885163/185918189-efcb309c-7dbd-48e6-8849-d49cb08ed475.jpg)
3. Also, I would remove token list dropdown and search for a token when connected to an unsupported network as they do not have any sense. WDYT?
![remove](https://user-images.githubusercontent.com/70885163/185918409-16fecf24-ff34-44eb-95a4-da637cd6aea3.jpg)
4. When I open Tokens table, and switch between networks for the 1st time, I see 'Unsupported network' [message](https://watch.screencastify.com/v/dTsf8jYpvtd37dwWIAsP). Maybe we should show a loader instead?

nenadV91 commented:
@elena 
1. fixed
2. I think just showing unsupported network is fine, otherwise we would show the same long text 2 times (in yellow popup also)
3. fixed
4. fixed
5. fixed
6. We can look into this more in the future but I think this is fine for now, COW and GNO being at the top and also tokens with balance

elena-zh commented:
Hey @nenadV91 , great job!
I have created #964 issue for Token lists and #965 issue for Unsupported network flow to address these issues later. 
As for the current changes, they LGTM. 
However, I have found 2 issues related to Gorli network: 
1. Balances are not loaded in Favourites tab there
2. I see a strange amount in Approved token
![image](https://user-images.githubusercontent.com/70885163/186174031-7e1df8a5-8f7b-45aa-ba54-a98024054d10.png)
 Could you please take a look at it? 

Thanks

nenadV91 commented:
@elena-zh Clear localstorage and try again, for me only USDT didn't work and it shouldn't even be there for Goerli since its not in the goerli list.
![Screenshot from 2022-08-24 14-39-43](https://user-images.githubusercontent.com/34926005/186420938-2c42c880-5948-4f46-9983-c5eb15443622.png)

elena-zh commented:
@nenadV91 , it did not help me. 
See the video: https://watch.screencastify.com/v/qRnFWYJZc8L7LaEmHyzF

Besides, an issue with COW approved amount is also relevant for me (you can see it on the video). 

Thanks!

elena-zh commented:
I assume that the issue might be related to tokens that are listed in Favourites, but are not in the existing and active token lists in the network. 

I mean if a user adds a custom token, and adds it to Favourites,
![image](https://user-images.githubusercontent.com/70885163/186449303-304a7ac4-0051-4281-b39d-104c987947a7.png)

![image](https://user-images.githubusercontent.com/70885163/186449188-6cc1b918-be78-494a-9806-621b1397199a.png)

 but this custom token does not exist in any list in the app, 
![image](https://user-images.githubusercontent.com/70885163/186449512-3e7e7059-166d-4cb7-8c2a-182b1467cc15.png)

so a user will face these issues with balances.

nenadV91 commented:
@elena-zh I've added a fix for fav token balance on #948
For the approved amount I can't reproduce it

elena-zh commented:
@nenadV91 , both issues are fixed in #948 
Thanks!

nenadV91 commented on src/custom/components/SearchModal/CommonBases/index.ts:
```diff
@@ -63,7 +63,7 @@ export const MobileWrapper = styled(AutoColumn)<{ showOverflow?: boolean }>`
       content: '';
       display: block;
-      background: linear-gradient(to top, #163861 0%, rgba(22, 56, 97, 0) 100%);
+      // background: linear-gradient(to top, #163861 0%, rgba(22, 56, 97, 0) 100%);
```
Yes this was causing a bug on light mode, and I'm not sure its needed.
